spark/src/main/scala/com/cloudera/livy/spark/client/ClientSession.scala

```diff
@@ -65,6 +71,14 @@ class ClientSession(val sessionId: Int, createRequest: CreateClientRequest)
     performOperation(job, false)
   }
 
+  def addFile(fileData: Array[Byte], fileName: String): Unit = {
```

Comment:
Why is the argument an `Array[Byte]`? Does that mean that if I try to upload a really large file it will load the whole thing in memory? Or fail? Neither option sounds really good.